### PR TITLE
[CPU] CPU plugin API 2.0 migration for IE_THROW and Exception

### PR DESCRIPTION
**Pull request by riverlijunjie**
Replace IE_THROW with OPENVINO_THROW

### Details:
 - IE_THROW() is replaced by OPENVINO_THROW
 - IE_THROW(NOTIMPLEMENT) is replaced by OPENVINO_THROW_NOT_IMPLEMENT
 - IE_ASSERT is replaced by OPENVINO_ASSERT

### Tickets:
 - 123984

### CODE REVIEW

**Comment by yuxu42** [comment]
@chenhu-wang Could you please take a review? Thanks!

**Comment by nshchego** [comment]
@riverlijunjie, could you please measure the binary size before and after this fix. It looks like the macro OPENVINO_THROW increase the binary size. If this is a significant amount, please create appropriate ticket for the Core team.

**Comment by riverlijunjie** [comment]
> @riverlijunjie, could you please measure the binary size before and after this fix. It looks like the macro OPENVINO_THROW increase the binary size. If this is a significant amount, please create appropriate ticket for the Core team.

Thanks @nshchego , this PR should only impact cpu plugin's binary size, so I checked cpu plugin's binary size of master branch and this PR, it shows that there is about 50KB ~160 KB binary size increasement for 3 different OS platforms:

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:dt="uuid:C2F41010-65B3-11d1-A29F-00AA00C14882"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=OneNote.File>
<meta name=Generator content="Microsoft OneNote 15">
</head>

<body lang=en-US style='font-family:Calibri;font-size:11.0pt'>


<div style='direction:ltr'>



  | master | PR_21005 | increased
-- | -- | -- | --
Windows | 33,840 KB | 33,893 KB | +53 KB
Ubuntu20.04 | 44,004 KB | 44,144 KB | +140 KB
MacOS | 45,957 KB | 46,117 KB | +160 KB



</div>


</body>

</html>

Do you think how many binary size increasing is accepted?

**Comment by nshchego** [comment]
> > @riverlijunjie, could you please measure the binary size before and after this fix. It looks like the macro OPENVINO_THROW increase the binary size. If this is a significant amount, please create appropriate ticket for the Core team.
> 
> Thanks @nshchego , this PR should only impact cpu plugin's binary size, so I checked cpu plugin's binary size of master branch and this PR, it shows that there is about 50KB ~160 KB binary size increasement for 3 different OS platforms:
> 
>  	master	PR_21005	increased
> Windows	33,840 KB	33,893 KB	+53 KB
> Ubuntu20.04	44,004 KB	44,144 KB	+140 KB
> MacOS	45,957 KB	46,117 KB	+160 KB
> Do you think how many binary size increasing is accepted?

We can merge that, but create appropriate ticket for this issue please.

**Comment by riverlijunjie** [comment]
> > > @riverlijunjie, could you please measure the binary size before and after this fix. It looks like the macro OPENVINO_THROW increase the binary size. If this is a significant amount, please create appropriate ticket for the Core team.
> > 
> > 
> > Thanks @nshchego , this PR should only impact cpu plugin's binary size, so I checked cpu plugin's binary size of master branch and this PR, it shows that there is about 50KB ~160 KB binary size increasement for 3 different OS platforms:
> >  	master	PR_21005	increased
> > Windows	33,840 KB	33,893 KB	+53 KB
> > Ubuntu20.04	44,004 KB	44,144 KB	+140 KB
> > MacOS	45,957 KB	46,117 KB	+160 KB
> > Do you think how many binary size increasing is accepted?
> 
> We can merge that, but create appropriate ticket for this issue please.

sure, create ticket: https://jira.devtools.intel.com/browse/CVS-124965

**Review by ilya-lavrenov — APPROVED** [review]

File: src/plugins/intel_cpu/src/config.cpp [hunk 1]
```diff
@@ -256,7 +256,7 @@ void Config::readProperties(const ov::AnyMap& prop, const ModelType modelType) {
                            << ". Supported values: PERFORMANCE, ACCURACY";
             }
         } else {
-            IE_THROW(NotFound) << "Unsupported property " << key << " by CPU plugin";
+            OPENVINO_THROW("NotFound: Unsupported property ", key, " by CPU plugin;");
```
Comment (chenhu-wang) [inline src/plugins/intel_cpu/src/config.cpp:259]: minor:
```suggestion
            OPENVINO_THROW("NotFound: Unsupported property ", key, " by CPU plugin.");
```

**Review by chenhu-wang — APPROVED** [review]

**Review by maxnick — APPROVED** [review]

File: src/plugins/intel_cpu/src/config.cpp (hunk 1 above)
Comment (riverlijunjie) [reply src/plugins/intel_cpu/src/config.cpp]: update it.

**Pull request merged**